CONTRIBUTING.md

```diff
@@ -2,21 +2,53 @@
 
 Observable Plot is released under the ISC license. You are welcome to fork this repository and to send us pull requests to contribute bug fixes or new features.
 
-To develop Observable Plot locally, clone this repository and install its dependencies:
+## Set up
+To set up your local development environment for Plot, follow these steps on your terminal:
 
-```
+```bash
+# 1. Clone this repository 
 git clone git@github.com:observablehq/plot.git
-cd plot
-yarn
+cd plot # cd into the folder
+
+# 2. Checkout a new development branch, by convention `user-name/feature-name`
+git checkout -b user-name/feature-name
+
+# 3. Install dependencies listed in the package.json file
+yarn 
 ```
 
+## Development
 Plot is written in ES modules and uses [Vite](https://vitejs.dev/) for development; this means that you can edit the Plot source code and examples, and they’ll update live as you save changes. To start Vite:
 
 ```
 yarn dev
 ```
 
-This should open http://localhost:8008/ where you can browse the tests.
+This should open http://localhost:8008/ in your browser where you can see the tests plots defined in `test/plots`. 
+
+![local development environment](img/localhost.png)
+
+If you'd like to create a new test plot as part of your development, you need to:
+
+1. Create a new `.js` file in `test/plots`, e.g., `test/plots/new-example.js` (an easy start is duplicating an existing example).
+2. Export the function from your `new-example.js` file in `test/index.js`. Because the first alphabetical option is displayed by on the local server, you may want to export the example as `_newExample`.
+
+### Development using Observable
+If you'd like to develop using Observable (or test your changes on an existing notebook), you'll need to first bundle your changes into a single file:
+
+```bash
+# Compile into a file `dist/plot.umd.js`
+yarn prepublishOnly
+```
+
+You can then load the local copy of Plot in an Observable notebook:
+
+```js
+// Load your copy into Observable
+Plot = require("http://localhost:8008/path/to/plot/dist/plot.umd.js")
```

Comment:
I don’t think we should recommend this approach specifically (because it doesn’t work in Safari), and more generally, I think we should not recommend that people “develop in Observable”. The recommended development pattern is to use Vite locally as described above. I think it’s okay (desirable, even) to describe how to _share a demo_ on Observable of some in-development feature. And in that case, you’d be uploading your built plot.umd.js as a file attachment and saying something like:

```js
Plot = require(await FileAttachment("plot.umd.js").url())
```

Reply:
I can appreciate that -- I'll remove this approach, and reframe the description about sharing on Observable.